src/sinks/pulsar/request_builder.rs

```diff
@@ -0,0 +1,76 @@
+use bytes::Bytes;
+use std::collections::HashMap;
+use std::io;
+use vector_common::finalization::EventFinalizers;
+use vector_common::request_metadata::RequestMetadata;
+
+use crate::sinks::pulsar::encoder::PulsarEncoder;
+use crate::sinks::pulsar::sink::PulsarEvent;
+use crate::sinks::util::metadata::RequestMetadataBuilder;
+use crate::sinks::util::request_builder::EncodeResult;
+use crate::sinks::util::{Compression, RequestBuilder};
+use crate::{
+    event::{Event, Finalizable},
+    sinks::pulsar::service::PulsarRequest,
+};
+
+#[derive(Clone)]
+pub(super) struct PulsarMetadata {
+    pub finalizers: EventFinalizers,
+    pub key: Option<Bytes>,
+    pub properties: Option<HashMap<String, Bytes>>,
+    pub timestamp_millis: Option<i64>,
+    pub topic: String,
+}
+
+pub(super) struct PulsarRequestBuilder {
+    pub(super) encoder: PulsarEncoder,
+}
+
+impl RequestBuilder<PulsarEvent> for PulsarRequestBuilder {
+    type Metadata = PulsarMetadata;
+    type Events = Event;
+    type Encoder = PulsarEncoder;
+    type Payload = Bytes;
+    type Request = PulsarRequest;
+    type Error = io::Error;
+
+    fn compression(&self) -> Compression {
+        // we don't use this compression from vector as Pulsar supports more
+        // compression formats
+        Compression::None
```

Comment:
A more relevant comment here might be that compression is handled by the Pulsar crate through the producer settings.

Reply:
added this